tests/warn/i15503g.scala

```diff
@@ -15,10 +15,17 @@ object Foo {
   private def g2(x: Int) = ??? // OK
 }
 
-package foo.test.i17101:
+object i17101:
   type Test[A] = A
   extension[A] (x: Test[A]) { // OK
     def value: A = x
     def causesIssue: Unit = println("oh no")
     def isAnIssue(y: A): Boolean = x == x // warn
   }
+
+object i23125:
+  trait Show:
+    def show(s: String) = s
+  extension (s: String)(using Show) // warn not used in repeat
+    def echo = println(summon[Show].show(s))
+    def repeat = s * 2
```

Comment:
it's OK if the warning is repeated for a second method. Alternatively, collect warnings by "site" and/or "origin" for helpful grouping. Maybe a future enhancement.